dbms/src/Common/HashTable/StringHashTable.h

```diff
@@ -296,13 +295,19 @@ class StringHashTable : private boost::noncopyable
             if ((reinterpret_cast<uintptr_t>(p) & 2048) == 0)
             {
                 memcpy(&n[0], p, 8);
-                n[0] &= -1ul >> s;
+                if constexpr (DB::isLittleEndian())
+                    n[0] &= (-1ULL >> s);
```

Comment:
it seem the old one use `-1ul`, while the new one use `-1ull`. is it expected?

Reply:
> 

Yes, it should be 8 bytes integer.